src/main/java/com/coveo/configuration/parameterstore/ParameterStorePropertySourceEnvironmentPostProcessor.java

```diff
@@ -29,15 +35,29 @@
     public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application)
     {
         if (!initialized && isParameterStorePropertySourceEnabled(environment)) {
-            environment.getPropertySources()
-                       .addFirst(new ParameterStorePropertySource(PARAMETER_STORE_PROPERTY_SOURCE_NAME,
-                                                                  new ParameterStoreSource(buildAWSSimpleSystemsManagementClient(environment),
-                                                                                           environment.getProperty(PARAMETER_STORE_HALT_BOOT_CONFIGURATION_PROPERTY,
-                                                                                                                   Boolean.class,
-                                                                                                                   Boolean.FALSE))));
-            if (!environment.getProperty(PARAMETER_STORE_SUPPORT_MULTIPLE_APPLICATION_CONTEXTS_CONFIGURATION_PROPERTY,
-                                         Boolean.class,
-                                         Boolean.FALSE)) {
+            boolean haltBoot = environment.getProperty(PARAMETER_STORE_HALT_BOOT_CONFIGURATION_PROPERTY,
+                                                       Boolean.class,
+                                                       Boolean.FALSE);
+            if (isMultiRegionEnabled(environment)) {
```

Comment:
I think just having multiples `signingRegions` would be enough here instead of an additional parameter no? I'm also thinking the multi-region and single-region could be in their own class for clarity purposes.

Reply:
Yeah. Thought about it and it probably would be much simpler. I added the property `multiRegion.enabled` to be more flexible. Multi region could be enabled without specifying the regions and it could fallback on `signingRegion` or the default one. But it's just too flexible. You don't need to enabled multi region if you don't have multiple regions. I'll change it to just `signingRegions`. I'll also put them in separate classes.